config/src/main/java/org/springframework/security/config/annotation/web/configurers/HeadersConfigurer.java

```diff
@@ -553,15 +553,15 @@ public PermissionsPolicyConfig permissionsPolicy() {
 	 * <p>
 	 * Configuration is provided to the {@link PermissionsPolicyHeaderWriter} which is
 	 * responsible for writing the header.
-	 * @return the {@link PermissionsPolicyConfig} for additional configuration
-	 * @throws IllegalArgumentException if policyDirectives is {@code null} or empty
+	 * </p>
+	 * @return the {@link HeadersConfigurer} for additional customizations
 	 * @since 5.5
 	 * @see PermissionsPolicyHeaderWriter
 	 */
-	public PermissionsPolicyConfig permissionsPolicy(Customizer<PermissionsPolicyConfig> permissionsPolicyCustomizer) {
+	public HeadersConfigurer<H> permissionsPolicy(Customizer<PermissionsPolicyConfig> permissionsPolicyCustomizer) {
```

Comment:
While certainly a bug, we cannot change the return type of a public method and remain binary compatible. And given that this has been in place for a few years now and that there is a simple workaround, I hesitate to make a change this aggressive.

Let's instead introduce a new method and deprecate this one. Perhaps `permissionsPolicyHeader` would be an appropriate name. We can remove `permissionsPolicy` in Spring Security 7.

Reply:
I understand your point of view but `permissionsPolicyHeader`  name doesn't seem to be coherent with other "DSL" methods remaining in Spring Security 7. Would you keep this incoherence in Spring Security 7 ? Unless you will rename other DSL methods with same convention.

Reply:
I will wait your thoughts :)

Reply:
@jzheaux have you seen my comments above?

Reply:
Hi @florianberthe. Yes, the method will be renamed from `permissionsPolicyHeader` to `permissionsPolicy` in Spring Security 7 when we remove the deprecated method.

Reply:
Hi, I will make this change later

Reply:
So the plan is to add a `permissionsPolicyHeader` method in Spring 6.3 with a correct return type?
Then in 7 the old `permissionsPolicy` method with incorrect return type will be removed and `permissionsPolicyHeader` will be renamed? 
That would mean that even if you now use the undeprecated `permissionsPolicyHeader` method in 6.3 you will have a breaking change when you upgrade to 7?
Or will the `permissionsPolicyHeader` method will also exist in 7 and be deprecated then?

Reply:
> So the plan is to add a permissionsPolicyHeader method in Spring 6.3 with a correct return type?

Yes, in Spring Security 6.4. As well as deprecating the current method with the wrong return type.

> Then in 7 the old permissionsPolicy method with incorrect return type will be removed and permissionsPolicyHeader will be renamed?

That's right.

> That would mean that even if you now use the undeprecated permissionsPolicyHeader method in 6.3 you will have a breaking change when you upgrade to 7?

Yes, a breaking change is fine since that will be a major version bump.

Reply:
Just my 2 cents but that does not seem entirely user friendly.

Yes in version 7 you can remove the deprecated method with incorrect return type. 
But instead of renaming the temporary method of version 6.4, deprecate that method in version 7.
Instead of removing the deprecated method it gets replaced by one with the correct return.

That way users of 6.4 have no breaking changes but can upgrade by searching deprecation.

So 6.4: 
```
@deprecated("Deprecated invalid return type, use temporary method below")
PermissionsPolicyConfig permissionsPolicy(..)

HeadersConfigurer<H> permissionsPolicyHeader(..)
```

And 7.0:
```
HeadersConfigurer<H> permissionsPolicy(..)

@deprecated("Use corrected method above")
HeadersConfigurer<H> permissionsPolicyHeader(..)
```